src/main/java/net/schmizz/sshj/common/KeyType.java

```diff
@@ -66,7 +66,7 @@ protected void writePubKeyContentsIntoBuffer(PublicKey pk, Buffer<?> buf) {
 
         @Override
         protected boolean isMyType(Key key) {
-            return (key instanceof RSAPublicKey || key instanceof RSAPrivateKey);
+            return "RSA".equals(key.getAlgorithm());
```

Comment:
Let's extract `"RSA"` and `"DSA"` into constants.

Reply:
I created a new class `KeyAlgorithm` in `common` that extracts all the constants used (implicitly) as arguments to `KeyFactory.getInstance(...)`.